modules/heat_conduction/src/interfacekernels/ThinLayerHeatTransfer.C

```diff
@@ -0,0 +1,115 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ThinLayerHeatTransfer.h"
+
+registerMooseObject("HeatConductionApp", ThinLayerHeatTransfer);
+
+InputParameters
+ThinLayerHeatTransfer::validParams()
+{
+  InputParameters params = InterfaceKernel::validParams();
+  params.addClassDescription("Model heat transfer across a thin domain with an interface.");
+  params.addParam<MaterialPropertyName>(
+      "specific_heat",
+      "specific_heat",
+      "Property name of the specific heat material property of the thin layer");
+
+  params.addParam<MaterialPropertyName>(
+      "density", "density", "Property name of the density material property of the thin layer");
+
+  params.addParam<MaterialPropertyName>(
+      "heat_source",
+      "heat_source",
+      "Property name of the heat source material property of the thin layer");
+
+  params.addParam<MaterialPropertyName>(
+      "thermal_conductivity", "thermal_conductivity", "Property name of the thermal conductivity");
+
+  params.addRequiredParam<Real>("thickness", "The thin layer thickness");
+  return params;
+}
+
+ThinLayerHeatTransfer::ThinLayerHeatTransfer(const InputParameters & parameters)
+  : InterfaceKernel(parameters),
+    _specific_heat(parameters.isParamSetByUser("specific_heat")
+                       ? getMaterialProperty<Real>("specific_heat")
+                       : getGenericZeroMaterialProperty<Real, false>()),
+    _density(parameters.isParamSetByUser("density")
+                 ? getMaterialProperty<Real>("density")
+                 : getGenericZeroMaterialProperty<Real, false>()),
+    _heat_source(parameters.isParamSetByUser("heat_source")
+                     ? getMaterialProperty<Real>("heat_source")
+                     : getGenericZeroMaterialProperty<Real, false>()),
```

Comment:
If a user relies on the default name and doesn't specify the parameter in the input file, then the `isParamSetByUser` logic would lead to retrieval of a zero property which may be a surprise to the user. I would suggest removing the default values and then using `isParamValid` instead of `isParamSetByUser`

Reply:
That is definitely better.